uint16 index
string name
string description